core/trino-main/src/main/java/io/trino/type/TypeCoercion.java

```diff
@@ -311,6 +311,7 @@ private TypeCompatibility typeCompatibilityForCovariantParametrizedType(Type fro
                 case StandardTypes.JSON:
                 case StandardTypes.INTERVAL_YEAR_TO_MONTH:
                 case StandardTypes.INTERVAL_DAY_TO_SECOND:
+                case StandardTypes.UUID:
```

Comment:
This seems unrelated to the first commit (StandardTypes.UUID existed before the change)
If this fixes something, can you please add a test?
Also, would it make sense to separate PR?